x-pack/plugin/slm/src/main/java/org/elasticsearch/xpack/slm/SnapshotRetentionTask.java

```diff
@@ -187,7 +187,7 @@ static Map<String, SnapshotLifecyclePolicy> getAllPoliciesWithRetentionEnabled(f
 
     static boolean snapshotEligibleForDeletion(
         SnapshotInfo snapshot,
-        Map<String, List<SnapshotInfo>> allSnapshots,
+        Map<String, Map<String, Map<SnapshotId, RepositoryData.SnapshotDetails>>> allSnapshotDetails,
```

Comment:
You're straining the very definition of data structures here, David. 😆

Reply:
If this was Haskell they'd be type synonyms.